presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -114,6 +114,7 @@
     private final Optional<TranslationMap> outerContext;
     private final Session session;
     private final SubqueryPlanner subqueryPlanner;
+    private final Map<NodeRef<Node>, RelationPlan> replacementMap;
```

Comment:
`replacementMap` is too vague of a name. Maybe rename to something more precise, such as "recursiveSubqueries".